### PR TITLE
clang-tidy: replace RefCounter with shared_ptr

### PR DESCRIPTION
**Pull request by neheb**
The latter is C++11.

Found with clang-analyzer-webkit.NoUncountedMemberChecker

### CODE REVIEW

**Comment by sbooth** [comment]
I think this change makes sense, thanks.

**Comment by neheb** [comment]
Rebased.

File: taglib/mpeg/mpegheader.h [hunk 1]
```diff
@@ -29,6 +29,8 @@
 #include "taglib.h"
 #include "taglib_export.h"
 
+#include <memory>
+
```
Comment (sbooth) [inline taglib/mpeg/mpegheader.h:33]: I noticed that `memory` is included here but not in most of the other headers. For consistency it should probably be included in all of them that use `shared_ptr`.

Comment (neheb) [reply taglib/mpeg/mpegheader.h:33]: Right. I did this as it seems most cop files use includes inherited from elsewhere.

Maybe include-what-you-use should be ran against the codebase.

Comment (sbooth) [reply taglib/mpeg/mpegheader.h:33]: That sounds like a great candidate for a PR once I get the current few merged.

Comment (neheb) [reply taglib/mpeg/mpegheader.h:33]: You still want me to add memory everywhere in this PR?

Comment (sbooth) [reply taglib/mpeg/mpegheader.h:33]: I think it's a good technique but not necessary for this PR. I'm not aware of a taglib style for includes.

**Comment by sbooth** [comment]
The only thing left for this PR that I know if is removing the `trefcounter` sources and adjusting the CMakeFiles appropriately.

**Comment by neheb** [comment]
Done.

**Pull request merged**